doc/DatabaseOperations.md

```diff
@@ -0,0 +1,154 @@
+$$;
+CREATE TABLE "prefsSafes" (
+    "prefsSafesId" VARCHAR(36) PRIMARY KEY NOT NULL,
+    "safeType" "prefsSafesType" NOT NULL,
+    name VARCHAR(64),
+    password VARCHAR(64),
+    email VARCHAR(32),
+    preferences JSONB NOT NULL DEFAULT '{}'
+);
+INSERT INTO "prefsSafes" ("prefsSafesId", "safeType", name, password, email)
+    VALUES ('prefsSafe-1', user, 'carla', 'null', 'someone@somewhere.com')
+    RETURNING *
+;
+
+```
+
+In both cases, the query is a single `String`.
+
+The returned `Promise` is configured with a rejection handler that logs any
+error.
+
+##### `bulkQuery(queryArray)` (Invoker)
+
+- `queryArray {Array}` An array of SQL statements.
+- Returns: `{Promise}` A promise whose values are the results of running the
+sequence of commands in the `queryArray`.
+
+Loop to execute the input array as a `fluid.promise.sequence()`.  The returned
+`Promise` is configured with a rejection handler that logs any error.
+
+While there is no constraint on the order of the SQL statements in the input
+array, this function can be used to define a logical sequence of database
+queries, where the order of the queries is defined by their position in the
+array.
+
+##### `loadFromJSON(tableName, records)` (Invoker)
+
+- `tableName {String}` The name of the table to load with the given records.
+- `jsonArray {Array}` An array of JSON objects containing the data to load into
+the given table.
+- Returns: `{Promise}` whose values is an array of successful INSERT results.
+
+Bulk load the given records into the given table.  The records' JSON field
+names must match the column names of the table, and the values of each field
+contain the value to store in the corresponding column.  For columns that are
+required and have no default value set, the JSON field must exist and its value
+be of the correct type and format for the column.  For table columns that are
+not required, or have a default value, the corresponding JSON field can be
+missing.
+
+##### `fluid.postgresdb.stringifyJoinKeys(keys)` (Function)
+
+- `keys {Array}` An array of strings as given by `Object.keys()`.
+- Returns: A comma-separated string of the keys, each quoted with double quotes.
+
+Utility to stringify each element of an array of object keys and join them
+as a single string separated by commas.  This is similar to the `Array.join()`
+function, but where each element is quoted with double quotes.
+
+##### `fluid.postgresdb.stringifyJoinValues(values)` (Function)
+
+- `values {Array}` Array of object values as given by `Object.values()`.
+- Returns: A comma-separated string of the values each quoted with single
+quotes.
+
+Utility to stringify each element of an array of object values and join them
+as a single string separated by commas.  This is similar to the `Array.join()`
+function, but where each element is quoted with a single quote.  Also, if
+a value is an object, it is stringified as well as quoted.
+
+##### `fluid.postgresdb.maybeStringify(aValue)` (Function)
```

Comment:
What's this API for? Questions I have for two data types it stringifies:

1. Object. When an object is given, would it be possible that the user wants to insert as a JSON data type rather than a string type?
2. Array. What the intention for replacing the square brackets with curly brackets?

Reply:
This API was part of the `loadFromJSON()` machinery, and is no longer used given @amb26's  [comment/suggestion below](https://github.com/fluid-project/preferencesServer/pull/6#discussion_r593512366).  As was said there, it's not advisable to create these strings by hand.  I've replaced the JSON-to-SQL translation code to use node-postgres-format.

Still, the purpose was to get the quotations and other special symbols correct for JSONB and Array column types.  The reason for the curly brackets is that is a way to INSERT into an array, for example:
`INSERT INTO "tableName" (roles) VALUES ('{user,admin}')";`.  Here the `roles` column is an array and the values are specified by '{user,admin}'.  Stringifying alone gives '[user,admin]', which is incorrect SQL.  However, that's all moot  since node-postgres-format does the conversion now.